### PR TITLE
Feature/add conversions

### PR DESCRIPTION
**Pull request by fivetran-jamie**
## PR Overview
**This PR will address the following Issue/Feature:**
No issue created

**This PR will result in the following new package version:**

v0.8.0 -- looking at our UI, `basic_ad_actions` is a child of `basic_ad` and is included by default with it, but juuuust in case someone doesn't have the new table, let's make it breaking, as it is a schema change.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## 🚨 Breaking Changes 🚨
- Incorporates the `basic_ad_actions` pre-built report in order to grab conversion data. `basic_ad_actions` is a child table of the already-required `basic_ad` report, broken down by `action_type`. 
  - Addition of the new `stg_facebook_ads__basic_ad_actions` (and its `_tmp` counterpart) staging model. Given that this is a schema change for the package, this a breaking change.

## Feature Updates
- With the inclusion of the `basic_ad_actions` source table, creates a `facebook_ads__basic_ad_actions_passthrough_metrics` variable to pass through additional conversion value metrics to downstream models. By default, the package includes only the conversion value calculated using the default attribution window, but your report may include calculations using the other windows defined [here](https://developers.facebook.com/docs/marketing-api/reference/ads-action-stats/). See [README](https://github.com/fivetran/dbt_facebook_ads_source/tree/main?tab=readme-ov-file#passing-through-additional-metrics) for details on how to use the new var.
- Adds `optimization_goal` field to `stg_facebook_ads__ad_set_history` model. This is defined as the optimization goal this ad set is using, possible values of which are defined [here](https://developers.facebook.com/docs/marketing-api/reference/ad-campaign/#fields).
- Adds `conversion_domain` field to `stg_facebook_ads__ad_history` model. This is defined as the domain you've configured the ad to convert to.

## Documentation
- Documents the ability to transform metrics provided to the `facebook_ads__basic_ad_passthrough_metrics` variable. See [README](https://github.com/fivetran/dbt_facebook_ads_source/tree/main?tab=readme-ov-file#passing-through-additional-metrics) for details.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present)

Before marking this PR as "ready for review" the following have been applied:
- [ ] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [ ] docs were regenerated (unless this PR does not include any code or yml updates) -- post-approval
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

See Validation tests in transform [PR](https://github.com/fivetran/dbt_facebook_ads/pull/43)

### CODE REVIEW

File: models/stg_facebook_ads__basic_ad_actions.sql
Comment (fivetran-joemarkiewicz) [inline models/stg_facebook_ads__basic_ad_actions.sql:43]: Should we consider making this a macro?

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:10]: Suggestion to not truncate variable

```suggestion
- With the inclusion of the `basic_ad_actions` source table, creates a `facebook_ads__basic_ad_actions_passthrough_metrics` variable to pass through additional conversion value metrics to downstream models. By default, the package includes only the conversion value calculated using the default attribution window, but your report may include calculations using the other windows defined [here](https://developers.facebook.com/docs/marketing-api/reference/ads-action-stats/). See [README](https://github.com/fivetran/dbt_facebook_ads_source/tree/main?tab=readme-ov-file#passing-through-additional-metrics) for details on how to use the new variable.
```

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:16]: Same note from transform, let's align on how to add Seer as a contributor here as well

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-jamie these changes look great! I just have a few minor suggestions, but none related to the actual code. I see this being a huge value add for our users!

Once you address the comments in this review let's hold off merging until the other updates in the Google and LinkedIn packages are ready as well. Thanks and great job!

File: models/stg_facebook_ads__basic_ad_actions.sql
Comment (fivetran-jamie) [reply models/stg_facebook_ads__basic_ad_actions.sql:43]: going back and forth on this one, as the code is more complex than what's usually in our staging models, but it is really only used in this one model, so creating a macro wouldn't make us any DRYer. what are your thoughts?

File: CHANGELOG.md
Comment (fivetran-jamie) [reply CHANGELOG.md:16]: added a section

File: models/stg_facebook_ads__basic_ad_actions.sql
Comment (fivetran-joemarkiewicz) [reply models/stg_facebook_ads__basic_ad_actions.sql:43]: That's a good point. If it won't make things much dryer then there is no point in wrapping it up in a macro. I am good leaving as is. Thanks!

**Comment by SwapfietsDataDaan** [comment]
We just ran into an issue where we wanted to create a solution for exactly this. This will be super helpful.

**Comment by fivetran-jamie** [comment]
FYI for anyone subscribed here, this is currently blocked. 

We are working with Engineering to ensure that the conversion monetary values (`action_values`) are included in a default report and therefore accessible to the package (currently, only the raw # of conversions are supported by the `basic_ad_actions` report).

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:6]:
```suggestion
- Incorporates the `basic_ad_actions` and `basic_ad_action_values` pre-built reports in order to grab conversion data. They are both child tables of the already-required `basic_ad` report, broken down by `action_type`. 
```

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:10]: Small rewording suggestion
```suggestion
- With the inclusion of the `basic_ad_actions` and `basic_ad_action_values` source tables, we have added the `facebook_ads__basic_ad_actions_passthrough_metrics` and `facebook_ads__basic_ad_action_values_passthrough_metrics` variables to pass through additional conversion value metrics to downstream models.
```

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:13]:
```suggestion
- Adds the `optimization_goal` field to `stg_facebook_ads__ad_set_history` model. This is defined as the optimization goal this ad set is using, possible values of which are defined [here](https://developers.facebook.com/docs/marketing-api/reference/ad-campaign/#fields).
- Adds the `conversion_domain` field to `stg_facebook_ads__ad_history` model. This is defined as the domain you've configured the ad to convert to.
```

File: docs/run_results.json
Comment (fivetran-joemarkiewicz) [inline docs/run_results.json:1]: Let's remove this file

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
Thanks @fivetran-jamie! This is looking really good. I have just a few comments and suggestions here as well as in the transform PR before this will be ready for final approval.

File: models/stg_facebook_ads__basic_ad_actions.sql
Comment (fivetran-jamie) [reply models/stg_facebook_ads__basic_ad_actions.sql:44]: this is now in 2 models, should i macro-fy?

Comment (fivetran-joemarkiewicz) [reply models/stg_facebook_ads__basic_ad_actions.sql:44]: @fivetran-jamie yeah since we need to maintain it in two separate places it makes sense to macro-fy. Thanks!

File: docs/run_results.json
Comment (fivetran-jamie) [reply docs/run_results.json:1]: removed

File: models/stg_facebook_ads__basic_ad_actions.sql
Comment (fivetran-joemarkiewicz) [reply models/stg_facebook_ads__basic_ad_actions.sql:44]: @fivetran-jamie can we macro-fy this or is it not entirely necessary?

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM just one comment bumping the macro-fy the code in the staging models.

File: models/stg_facebook_ads__basic_ad_actions.sql
Comment (fivetran-jamie) [reply models/stg_facebook_ads__basic_ad_actions.sql]: added!

File: CHANGELOG.md
Comment (fivetran-reneeli) [inline CHANGELOG.md:7]:
```suggestion
  - Addition of the new `stg_facebook_ads__basic_ad_actions` and `stg_facebook_ads__basic_ad_action_values` (and their `_tmp` counterparts) staging models. Given that this is a schema change for the package, this is a breaking change.
```

File: CHANGELOG.md
Comment (fivetran-reneeli) [inline CHANGELOG.md:11]: very minor but do we want to just mention what the default attribution window is

**Review by fivetran-reneeli — APPROVED** [review]
very minor documentation comments! approved!

File: README.md
Comment (fivetran-reneeli) [inline README.md:80]: also stating what the default is here?

File: CHANGELOG.md
Comment (fivetran-jamie) [reply CHANGELOG.md:11]: adding a link to our connector docs about it

File: README.md
Comment (fivetran-jamie) [reply README.md:80]: adding link

**Pull request merged**
